src/ImageSharp/Image/ImageFrameCollection.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp
+{
+    /// <summary>
+    /// Encapsulates an imaged collection of frames.
+    /// </summary>
+    /// <typeparam name="TPixel">The type of the pixel.</typeparam>
+    public sealed class ImageFrameCollection<TPixel> : IEnumerable<ImageFrame<TPixel>>, IDisposable
+        where TPixel : struct, IPixel<TPixel>
+    {
+        private readonly IList<ImageFrame<TPixel>> frames = new List<ImageFrame<TPixel>>();
+
+        internal ImageFrameCollection(int width, int height)
+        {
+            this.Add(new ImageFrame<TPixel>(width, height));
+        }
+
+        internal ImageFrameCollection(IEnumerable<ImageFrame<TPixel>> frames)
+        {
+            Guard.NotNullOrEmpty(frames, nameof(frames));
+            foreach (ImageFrame<TPixel> f in frames)
+            {
+                this.Add(f);
+            }
+        }
+
+        /// <summary>
+        /// Gets the count.
+        /// </summary>
+        public int Count => this.frames.Count;
+
+        /// <summary>
+        /// Gets the root frame.
+        /// </summary>
+        public ImageFrame<TPixel> RootFrame => this.frames.Count > 0 ? this.frames[0] : null;
```

Comment:
Is it valid to have 0 frames? Shouldn't we throw in the constructor instead?

Reply:
yes you can have zero frames, it will have zero when the ImageFrameCollection as been disposed.

Reply:
Q1: Shouldn't we throw `ObjectDisposedException` instead of returning null instead?
Q2: Looks like it's possible to dispose an `ImageFrameCollection` without disposing the underlying image bringing `Image<T>` into an invalid state. Shouldn't we drop `IDisposable` from `ImageFrameCollection<T>` API and control it directly from `Image<T>`?

Reply:
1. yes we should probably throw an `ObjectDisposedException`.

2. I don;t think its really an issue that you can dispose of the ImageFrameCollection without disposing of the image. You can do the same for ImageFrame too.  I suppose we could have the Image expose and interface of `IImageFrameCollection` that doesn't support `IDisposeable` then the users can't call dispose of the collection separately to the Image.